Home screen Top Stores Section

M-Abdelnabi opened the pull request:
### This PR adds the top stores section

> enhance the store card and fix some issues 
> add optional discount badge on the top left edge of the card
> add callback functions to see all top stores and see specific store details
> this section is reusable we could use it as a separated screen to show all top stores

## Screenshots/Videos (if applicable)
![WhatsApp Image 2025-10-27 at 6 49 58 AM](https://github.com/user-attachments/assets/f00fa050-e65b-4cd8-a1a1-60e62f3e03ff)

M-Abdelnabi committed:
fix favourite icon in store card

M-Abdelnabi committed:
add discount tag widget and its icons

M-Abdelnabi committed:
use discount tag with store card

M-Abdelnabi committed:
refactor discount tag widget

M-Abdelnabi committed:
rename icon tag

M-Abdelnabi committed:
refactor discount tag inside the store card

M-Abdelnabi committed:
use nested containers to simplify the usage of padding

M-Abdelnabi committed:
use nested containers to simplify the usage of padding

M-Abdelnabi committed:
make the discount tag optional

M-Abdelnabi committed:
add card click option

M-Abdelnabi committed:
add card click option

M-Abdelnabi committed:
add top stores section

M-Abdelnabi committed:
handle card clicks

M-Abdelnabi committed:
remove unnecessary padding

M-Abdelnabi committed:
remove unnecessary padding from section header

M-Abdelnabi committed:
refactor store card

baraa0abd commented on lib/core/design_system/widgets/cards/store_card.dart:
```diff
@@ -2,80 +2,103 @@ import 'package:flutter/material.dart';
+                              Colors.transparent,
+                              Color(0x00000000),
+                              Color(0xFF000000),
+                            ],
```
please use design system colors

abdo-essam committed:
Merge branch 'refs/heads/feature/home-screen-design' into homeScreen/top-stores-section

abdo-essam committed:
feat(ui): Add top stores section to home screen

Adds TopStoresSection widget to display a list of stores
with images, names, and discounts. Includes StoreCard
widget and supporting assets.
